src/auth/auth-service.ts

```diff
@@ -36,3 +37,14 @@ export const getToken = async () => {
   await kc.updateToken(5);
   return kc.token;
 };
+
+//expand authorities protocol to object structure
+const extractResourceRoles = (authorities: string) =>
+  authorities
+    .split(',')
+    .map(authorityDescriptor => authorityDescriptor.split(':'))
+    .map(parts => ({ resource: parts[0], resourceId: parts[1], role: parts[2] }));
+
+//check if user has any role in selected resource
+export const hasPermission: (string, resourceId) => boolean = (resource, resourceId) =>
+  !!find(extractResourceRoles(get(kc.tokenParsed, 'authorities')), { resource, resourceId });
```

Comment:
Is it possible to make this easier to follow? Not very clear as to what is going on here.

Reply:
det var før ;) 
noen ganger lønner dert ikke seg å fjerne navner og ha enlinjer